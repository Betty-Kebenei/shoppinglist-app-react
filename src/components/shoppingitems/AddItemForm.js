import React from 'react';
import {
    FormGroup, 
    ControlLabel,
    FormControl,
    Button 
} from 'react-bootstrap';

const AddItemForm = (props) => {

    const {
        itemname,
        quantity,
        units,
        price,
        currency,
        onChange,
        onSubmit
    } = props;

    // Return an add item react-bootstrap form.
    return (
        <div>
            <form onSubmit={onSubmit}>
                <FormGroup
                    controlId="Itemname"
                    >
                    <ControlLabel>Itemname</ControlLabel>
                        <FormControl
                            type="text"
                            name="itemname"
                            value={itemname}
                            placeholder="Enter itemname"
                            onChange={onChange}
                            required
                        />
                </FormGroup>
                <FormGroup
                    controlId="Quantity"
                    >
                    <ControlLabel>Quantity</ControlLabel>
                        <FormControl
                            type="text"
                            name="quantity"
                            value={quantity}
                            placeholder="Enter quantity"
                            onChange={onChange}
                        />
                </FormGroup>
                <FormGroup
                    controlId="Units"
                    >
                    <ControlLabel>Units of Measurement</ControlLabel>
                        <FormControl
                            type="text"
                            name="units"
                            value={units}
                            placeholder="Enter units"
                            onChange={onChange}
                        />
                </FormGroup>
                <FormGroup
                    controlId="Price"
                    >
                    <ControlLabel>Price</ControlLabel>
                        <FormControl
                            type="text"
                            name="price"
                            value={price}
                            placeholder="Enter price"
                            onChange={onChange}
                        />
                </FormGroup>
                <FormGroup
                    controlId="currency"
                    >
                    <ControlLabel>Currency</ControlLabel>
                        <FormControl
                            type="text"
                            name="currency"
                            value={currency}
                            placeholder="Enter currency"
                            onChange={onChange}
                        />
                </FormGroup>
                <Button type="submit">Submit</Button>
                <FormControl.Feedback />
            </form>   
        </div>
    );
} 
export default AddItemForm;