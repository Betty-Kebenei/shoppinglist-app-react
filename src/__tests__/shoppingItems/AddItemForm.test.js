import React from 'react';
import { shallow, mount, render } from 'enzyme';

import Config from '../../Config';

import AddItemForm from '../../components/shoppingitems/AddItemForm';

describe('interface of AddItemForm component', () => {
    it('should render', () => {
        const wrapper = shallow(<AddItemForm />);
        expect(wrapper).toMatchSnapshot();
    });
});