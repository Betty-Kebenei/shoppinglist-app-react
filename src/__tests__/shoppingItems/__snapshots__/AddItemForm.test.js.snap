// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`interface of AddItemForm component should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AddItemForm />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <form
        onSubmit={undefined}
      >
        <FormGroup
          bsClass="form-group"
          controlId="Itemname"
        >
          <ControlLabel
            bsClass="control-label"
            srOnly={false}
          >
            Itemname
          </ControlLabel>
          <FormControl
            bsClass="form-control"
            componentClass="input"
            name="itemname"
            onChange={undefined}
            placeholder="Enter itemname"
            required={true}
            type="text"
            value={undefined}
          />
        </FormGroup>
        <FormGroup
          bsClass="form-group"
          controlId="Quantity"
        >
          <ControlLabel
            bsClass="control-label"
            srOnly={false}
          >
            Quantity
          </ControlLabel>
          <FormControl
            bsClass="form-control"
            componentClass="input"
            name="quantity"
            onChange={undefined}
            placeholder="Enter quantity"
            type="text"
            value={undefined}
          />
        </FormGroup>
        <FormGroup
          bsClass="form-group"
          controlId="Units"
        >
          <ControlLabel
            bsClass="control-label"
            srOnly={false}
          >
            Units of Measurement
          </ControlLabel>
          <FormControl
            bsClass="form-control"
            componentClass="input"
            name="units"
            onChange={undefined}
            placeholder="Enter units"
            type="text"
            value={undefined}
          />
        </FormGroup>
        <FormGroup
          bsClass="form-group"
          controlId="Price"
        >
          <ControlLabel
            bsClass="control-label"
            srOnly={false}
          >
            Price
          </ControlLabel>
          <FormControl
            bsClass="form-control"
            componentClass="input"
            name="price"
            onChange={undefined}
            placeholder="Enter price"
            type="text"
            value={undefined}
          />
        </FormGroup>
        <FormGroup
          bsClass="form-group"
          controlId="currency"
        >
          <ControlLabel
            bsClass="control-label"
            srOnly={false}
          >
            Currency
          </ControlLabel>
          <FormControl
            bsClass="form-control"
            componentClass="input"
            name="currency"
            onChange={undefined}
            placeholder="Enter currency"
            type="text"
            value={undefined}
          />
        </FormGroup>
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="default"
          disabled={false}
          type="submit"
        >
          Submit
        </Button>
        <FormControlFeedback
          bsClass="form-control-feedback"
          bsRole="feedback"
        />
      </form>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <FormGroup
            bsClass="form-group"
            controlId="Itemname"
          >
            <ControlLabel
              bsClass="control-label"
              srOnly={false}
            >
              Itemname
            </ControlLabel>
            <FormControl
              bsClass="form-control"
              componentClass="input"
              name="itemname"
              onChange={undefined}
              placeholder="Enter itemname"
              required={true}
              type="text"
              value={undefined}
            />
          </FormGroup>,
          <FormGroup
            bsClass="form-group"
            controlId="Quantity"
          >
            <ControlLabel
              bsClass="control-label"
              srOnly={false}
            >
              Quantity
            </ControlLabel>
            <FormControl
              bsClass="form-control"
              componentClass="input"
              name="quantity"
              onChange={undefined}
              placeholder="Enter quantity"
              type="text"
              value={undefined}
            />
          </FormGroup>,
          <FormGroup
            bsClass="form-group"
            controlId="Units"
          >
            <ControlLabel
              bsClass="control-label"
              srOnly={false}
            >
              Units of Measurement
            </ControlLabel>
            <FormControl
              bsClass="form-control"
              componentClass="input"
              name="units"
              onChange={undefined}
              placeholder="Enter units"
              type="text"
              value={undefined}
            />
          </FormGroup>,
          <FormGroup
            bsClass="form-group"
            controlId="Price"
          >
            <ControlLabel
              bsClass="control-label"
              srOnly={false}
            >
              Price
            </ControlLabel>
            <FormControl
              bsClass="form-control"
              componentClass="input"
              name="price"
              onChange={undefined}
              placeholder="Enter price"
              type="text"
              value={undefined}
            />
          </FormGroup>,
          <FormGroup
            bsClass="form-group"
            controlId="currency"
          >
            <ControlLabel
              bsClass="control-label"
              srOnly={false}
            >
              Currency
            </ControlLabel>
            <FormControl
              bsClass="form-control"
              componentClass="input"
              name="currency"
              onChange={undefined}
              placeholder="Enter currency"
              type="text"
              value={undefined}
            />
          </FormGroup>,
          <Button
            active={false}
            block={false}
            bsClass="btn"
            bsStyle="default"
            disabled={false}
            type="submit"
          >
            Submit
          </Button>,
          <FormControlFeedback
            bsClass="form-control-feedback"
            bsRole="feedback"
          />,
        ],
        "onSubmit": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "form-group",
            "children": Array [
              <ControlLabel
                bsClass="control-label"
                srOnly={false}
              >
                Itemname
              </ControlLabel>,
              <FormControl
                bsClass="form-control"
                componentClass="input"
                name="itemname"
                onChange={undefined}
                placeholder="Enter itemname"
                required={true}
                type="text"
                value={undefined}
              />,
            ],
            "controlId": "Itemname",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "bsClass": "control-label",
                "children": "Itemname",
                "srOnly": false,
              },
              "ref": null,
              "rendered": "Itemname",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "bsClass": "form-control",
                "componentClass": "input",
                "name": "itemname",
                "onChange": undefined,
                "placeholder": "Enter itemname",
                "required": true,
                "type": "text",
                "value": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "form-group",
            "children": Array [
              <ControlLabel
                bsClass="control-label"
                srOnly={false}
              >
                Quantity
              </ControlLabel>,
              <FormControl
                bsClass="form-control"
                componentClass="input"
                name="quantity"
                onChange={undefined}
                placeholder="Enter quantity"
                type="text"
                value={undefined}
              />,
            ],
            "controlId": "Quantity",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "bsClass": "control-label",
                "children": "Quantity",
                "srOnly": false,
              },
              "ref": null,
              "rendered": "Quantity",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "bsClass": "form-control",
                "componentClass": "input",
                "name": "quantity",
                "onChange": undefined,
                "placeholder": "Enter quantity",
                "type": "text",
                "value": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "form-group",
            "children": Array [
              <ControlLabel
                bsClass="control-label"
                srOnly={false}
              >
                Units of Measurement
              </ControlLabel>,
              <FormControl
                bsClass="form-control"
                componentClass="input"
                name="units"
                onChange={undefined}
                placeholder="Enter units"
                type="text"
                value={undefined}
              />,
            ],
            "controlId": "Units",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "bsClass": "control-label",
                "children": "Units of Measurement",
                "srOnly": false,
              },
              "ref": null,
              "rendered": "Units of Measurement",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "bsClass": "form-control",
                "componentClass": "input",
                "name": "units",
                "onChange": undefined,
                "placeholder": "Enter units",
                "type": "text",
                "value": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "form-group",
            "children": Array [
              <ControlLabel
                bsClass="control-label"
                srOnly={false}
              >
                Price
              </ControlLabel>,
              <FormControl
                bsClass="form-control"
                componentClass="input"
                name="price"
                onChange={undefined}
                placeholder="Enter price"
                type="text"
                value={undefined}
              />,
            ],
            "controlId": "Price",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "bsClass": "control-label",
                "children": "Price",
                "srOnly": false,
              },
              "ref": null,
              "rendered": "Price",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "bsClass": "form-control",
                "componentClass": "input",
                "name": "price",
                "onChange": undefined,
                "placeholder": "Enter price",
                "type": "text",
                "value": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "form-group",
            "children": Array [
              <ControlLabel
                bsClass="control-label"
                srOnly={false}
              >
                Currency
              </ControlLabel>,
              <FormControl
                bsClass="form-control"
                componentClass="input"
                name="currency"
                onChange={undefined}
                placeholder="Enter currency"
                type="text"
                value={undefined}
              />,
            ],
            "controlId": "currency",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "bsClass": "control-label",
                "children": "Currency",
                "srOnly": false,
              },
              "ref": null,
              "rendered": "Currency",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "bsClass": "form-control",
                "componentClass": "input",
                "name": "currency",
                "onChange": undefined,
                "placeholder": "Enter currency",
                "type": "text",
                "value": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "active": false,
            "block": false,
            "bsClass": "btn",
            "bsStyle": "default",
            "children": "Submit",
            "disabled": false,
            "type": "submit",
          },
          "ref": null,
          "rendered": "Submit",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "form-control-feedback",
            "bsRole": "feedback",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "form",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form
          onSubmit={undefined}
        >
          <FormGroup
            bsClass="form-group"
            controlId="Itemname"
          >
            <ControlLabel
              bsClass="control-label"
              srOnly={false}
            >
              Itemname
            </ControlLabel>
            <FormControl
              bsClass="form-control"
              componentClass="input"
              name="itemname"
              onChange={undefined}
              placeholder="Enter itemname"
              required={true}
              type="text"
              value={undefined}
            />
          </FormGroup>
          <FormGroup
            bsClass="form-group"
            controlId="Quantity"
          >
            <ControlLabel
              bsClass="control-label"
              srOnly={false}
            >
              Quantity
            </ControlLabel>
            <FormControl
              bsClass="form-control"
              componentClass="input"
              name="quantity"
              onChange={undefined}
              placeholder="Enter quantity"
              type="text"
              value={undefined}
            />
          </FormGroup>
          <FormGroup
            bsClass="form-group"
            controlId="Units"
          >
            <ControlLabel
              bsClass="control-label"
              srOnly={false}
            >
              Units of Measurement
            </ControlLabel>
            <FormControl
              bsClass="form-control"
              componentClass="input"
              name="units"
              onChange={undefined}
              placeholder="Enter units"
              type="text"
              value={undefined}
            />
          </FormGroup>
          <FormGroup
            bsClass="form-group"
            controlId="Price"
          >
            <ControlLabel
              bsClass="control-label"
              srOnly={false}
            >
              Price
            </ControlLabel>
            <FormControl
              bsClass="form-control"
              componentClass="input"
              name="price"
              onChange={undefined}
              placeholder="Enter price"
              type="text"
              value={undefined}
            />
          </FormGroup>
          <FormGroup
            bsClass="form-group"
            controlId="currency"
          >
            <ControlLabel
              bsClass="control-label"
              srOnly={false}
            >
              Currency
            </ControlLabel>
            <FormControl
              bsClass="form-control"
              componentClass="input"
              name="currency"
              onChange={undefined}
              placeholder="Enter currency"
              type="text"
              value={undefined}
            />
          </FormGroup>
          <Button
            active={false}
            block={false}
            bsClass="btn"
            bsStyle="default"
            disabled={false}
            type="submit"
          >
            Submit
          </Button>
          <FormControlFeedback
            bsClass="form-control-feedback"
            bsRole="feedback"
          />
        </form>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <FormGroup
              bsClass="form-group"
              controlId="Itemname"
            >
              <ControlLabel
                bsClass="control-label"
                srOnly={false}
              >
                Itemname
              </ControlLabel>
              <FormControl
                bsClass="form-control"
                componentClass="input"
                name="itemname"
                onChange={undefined}
                placeholder="Enter itemname"
                required={true}
                type="text"
                value={undefined}
              />
            </FormGroup>,
            <FormGroup
              bsClass="form-group"
              controlId="Quantity"
            >
              <ControlLabel
                bsClass="control-label"
                srOnly={false}
              >
                Quantity
              </ControlLabel>
              <FormControl
                bsClass="form-control"
                componentClass="input"
                name="quantity"
                onChange={undefined}
                placeholder="Enter quantity"
                type="text"
                value={undefined}
              />
            </FormGroup>,
            <FormGroup
              bsClass="form-group"
              controlId="Units"
            >
              <ControlLabel
                bsClass="control-label"
                srOnly={false}
              >
                Units of Measurement
              </ControlLabel>
              <FormControl
                bsClass="form-control"
                componentClass="input"
                name="units"
                onChange={undefined}
                placeholder="Enter units"
                type="text"
                value={undefined}
              />
            </FormGroup>,
            <FormGroup
              bsClass="form-group"
              controlId="Price"
            >
              <ControlLabel
                bsClass="control-label"
                srOnly={false}
              >
                Price
              </ControlLabel>
              <FormControl
                bsClass="form-control"
                componentClass="input"
                name="price"
                onChange={undefined}
                placeholder="Enter price"
                type="text"
                value={undefined}
              />
            </FormGroup>,
            <FormGroup
              bsClass="form-group"
              controlId="currency"
            >
              <ControlLabel
                bsClass="control-label"
                srOnly={false}
              >
                Currency
              </ControlLabel>
              <FormControl
                bsClass="form-control"
                componentClass="input"
                name="currency"
                onChange={undefined}
                placeholder="Enter currency"
                type="text"
                value={undefined}
              />
            </FormGroup>,
            <Button
              active={false}
              block={false}
              bsClass="btn"
              bsStyle="default"
              disabled={false}
              type="submit"
            >
              Submit
            </Button>,
            <FormControlFeedback
              bsClass="form-control-feedback"
              bsRole="feedback"
            />,
          ],
          "onSubmit": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "form-group",
              "children": Array [
                <ControlLabel
                  bsClass="control-label"
                  srOnly={false}
                >
                  Itemname
                </ControlLabel>,
                <FormControl
                  bsClass="form-control"
                  componentClass="input"
                  name="itemname"
                  onChange={undefined}
                  placeholder="Enter itemname"
                  required={true}
                  type="text"
                  value={undefined}
                />,
              ],
              "controlId": "Itemname",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "bsClass": "control-label",
                  "children": "Itemname",
                  "srOnly": false,
                },
                "ref": null,
                "rendered": "Itemname",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "bsClass": "form-control",
                  "componentClass": "input",
                  "name": "itemname",
                  "onChange": undefined,
                  "placeholder": "Enter itemname",
                  "required": true,
                  "type": "text",
                  "value": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "form-group",
              "children": Array [
                <ControlLabel
                  bsClass="control-label"
                  srOnly={false}
                >
                  Quantity
                </ControlLabel>,
                <FormControl
                  bsClass="form-control"
                  componentClass="input"
                  name="quantity"
                  onChange={undefined}
                  placeholder="Enter quantity"
                  type="text"
                  value={undefined}
                />,
              ],
              "controlId": "Quantity",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "bsClass": "control-label",
                  "children": "Quantity",
                  "srOnly": false,
                },
                "ref": null,
                "rendered": "Quantity",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "bsClass": "form-control",
                  "componentClass": "input",
                  "name": "quantity",
                  "onChange": undefined,
                  "placeholder": "Enter quantity",
                  "type": "text",
                  "value": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "form-group",
              "children": Array [
                <ControlLabel
                  bsClass="control-label"
                  srOnly={false}
                >
                  Units of Measurement
                </ControlLabel>,
                <FormControl
                  bsClass="form-control"
                  componentClass="input"
                  name="units"
                  onChange={undefined}
                  placeholder="Enter units"
                  type="text"
                  value={undefined}
                />,
              ],
              "controlId": "Units",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "bsClass": "control-label",
                  "children": "Units of Measurement",
                  "srOnly": false,
                },
                "ref": null,
                "rendered": "Units of Measurement",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "bsClass": "form-control",
                  "componentClass": "input",
                  "name": "units",
                  "onChange": undefined,
                  "placeholder": "Enter units",
                  "type": "text",
                  "value": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "form-group",
              "children": Array [
                <ControlLabel
                  bsClass="control-label"
                  srOnly={false}
                >
                  Price
                </ControlLabel>,
                <FormControl
                  bsClass="form-control"
                  componentClass="input"
                  name="price"
                  onChange={undefined}
                  placeholder="Enter price"
                  type="text"
                  value={undefined}
                />,
              ],
              "controlId": "Price",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "bsClass": "control-label",
                  "children": "Price",
                  "srOnly": false,
                },
                "ref": null,
                "rendered": "Price",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "bsClass": "form-control",
                  "componentClass": "input",
                  "name": "price",
                  "onChange": undefined,
                  "placeholder": "Enter price",
                  "type": "text",
                  "value": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "form-group",
              "children": Array [
                <ControlLabel
                  bsClass="control-label"
                  srOnly={false}
                >
                  Currency
                </ControlLabel>,
                <FormControl
                  bsClass="form-control"
                  componentClass="input"
                  name="currency"
                  onChange={undefined}
                  placeholder="Enter currency"
                  type="text"
                  value={undefined}
                />,
              ],
              "controlId": "currency",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "bsClass": "control-label",
                  "children": "Currency",
                  "srOnly": false,
                },
                "ref": null,
                "rendered": "Currency",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "bsClass": "form-control",
                  "componentClass": "input",
                  "name": "currency",
                  "onChange": undefined,
                  "placeholder": "Enter currency",
                  "type": "text",
                  "value": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "active": false,
              "block": false,
              "bsClass": "btn",
              "bsStyle": "default",
              "children": "Submit",
              "disabled": false,
              "type": "submit",
            },
            "ref": null,
            "rendered": "Submit",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "form-control-feedback",
              "bsRole": "feedback",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "form",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
